core/src/main/scala/kafka/log/Log.scala

```diff
@@ -556,13 +563,13 @@ class Log(@volatile var dir: File,
   def numberOfSegments: Int = segments.size
 
   /**
-   * Close this log
+   * Close this log.
+   * File handlers of this log will be left open so that the log can be deleted after it is closed.
```

Comment:
Is this comment correct? It seems that logSegment.close() does a superset of logSegment.closeHandlers().

Reply:
I think it is correct. I verified that close() does not set AbstractIndex.mmap to null. Note that close() can not set mmap to null so that AbstractIndex.delete() can be executed by the scheduler thread.

Reply:
Ok. Then "File handlers" can be a bit confusing since we do close the channel associated with the log segment. How about changing the comment to the following?

"The memory mapped buffer for index files of this log will be left open until the log is deleted"

Reply:
Sure. I have updated the comment and related method names as suggested.